c2_components/src/mfx_c2_decoder_component.cpp

```diff
@@ -829,16 +837,25 @@ mfxStatus MfxC2DecoderComponent::InitDecoder(std::shared_ptr<C2BlockPool> c2_all
         }
 
         if (MFX_ERR_NONE == mfx_res) {
+            // set memory type according to consumer usage sent from framework
+            m_mfxVideoParams.IOPattern = (C2MemoryUsage::CPU_READ == m_consumerUsage) ?
+                    MFX_IOPATTERN_OUT_SYSTEM_MEMORY : MFX_IOPATTERN_OUT_VIDEO_MEMORY;
+            MFX_DEBUG_TRACE_I32(m_mfxVideoParams.IOPattern);
+
             // Query required surfaces number for decoder
             mfxFrameAllocRequest decRequest = {};
             mfx_res = m_mfxDecoder->QueryIOSurf(&m_mfxVideoParams, &decRequest);
             if (MFX_ERR_NONE == mfx_res) {
-               if (m_uOutputDelay < decRequest.NumFrameSuggested) {
+                if (m_uOutputDelay < decRequest.NumFrameSuggested) {
                     MFX_DEBUG_TRACE_MSG("More buffer needed for decoder output!");
                     ALOGE("More buffer needed for decoder output! Actual: %d. Expected: %d",
                         m_uOutputDelay, decRequest.NumFrameSuggested);
                     mfx_res = MFX_ERR_MORE_SURFACE;
-               }
+                }
+                m_surfaceNum = MFX_MAX(decRequest.NumFrameSuggested, 4);
```

Comment:
so at least 4 surface needed?

Reply:
Yes, for system memory, as there is memory copy, MSDK doesn't need to lock the output surface (such as DPB). After calling MFXVideoCORE_SyncOperation, the output surface is in free state.

Reply:
Ok, Thanks